test/credential-test.js

```diff
@@ -156,6 +156,61 @@ test('verify with empty password', function(t) {
 
 });
 
+test('constantEquals', function (t) {
+  var ctc = require('../constantTimeCompare');
+
+  function timed_compare(a, b) {
+    var start = process.hrtime();
+    ctc(a, b);
+    return process.hrtime(start)[1];
+  }
+  var i,
+      iterations = 5000,
```

Comment:
Yeah, the optimization/priming is a problem.

That said, I suspect that if enough runs are called, and the call result (equal,inequal,difflen) are randomized, then the proper stats method will treat the optimization priming as noise.